app/components/MainScreen/ProjectAndSiteSelector.tsx

```diff
@@ -0,0 +1,182 @@
+import bbox from '@turf/bbox';
+import turfCenter from '@turf/center';
+import React, { useEffect } from 'react';
+import { StyleSheet, View } from 'react-native';
+import DropDownPicker from 'react-native-dropdown-picker';
+import { Colors, Typography } from '../../styles';
+
+interface Props {
+  projects: any[];
+  showProjectOptions: boolean;
+  setShowProjectOptions: React.Dispatch<React.SetStateAction<boolean>>;
+  showProjectSiteOptions: boolean;
+  setShowProjectSiteOptions: React.Dispatch<React.SetStateAction<boolean>>;
+  setSiteCenterCoordinate: React.Dispatch<React.SetStateAction<any>>;
+  setSiteBounds: React.Dispatch<React.SetStateAction<any>>;
+  projectSites: any;
+  setProjectSites: React.Dispatch<React.SetStateAction<any>>;
+}
+
+const ProjectAndSiteSelector = ({
+  projects,
+  showProjectOptions,
+  setShowProjectOptions,
+  showProjectSiteOptions,
+  setShowProjectSiteOptions,
+  setSiteCenterCoordinate,
+  setSiteBounds,
+  projectSites,
+  setProjectSites,
+}: Props) => {
+  const [projectOptions, setProjectOptions] = React.useState([]);
+
+  // used to set the selected project
+  const [selectedProjectId, setSelectedProjectId] = React.useState(null);
+  // used to set the selected project site
+  const [selectedProjectSiteId, setSelectedProjectSiteId] = React.useState(null);
+
+  useEffect(() => {
+    const options = projects.map(project => ({
+      label: project.name,
+      value: project.id,
+      sites: project.sites,
+    }));
+
+    if (options.length > 0) {
+      setSelectedProjectId(options[0].value);
+      setProjectSitesUsingProject(options[0]);
+    }
+    setProjectOptions(options);
+  }, [projects]);
+
+  useEffect(() => {
+    if (selectedProjectId) {
+      const project = projects.find(project => project.id === selectedProjectId);
+      const sites = project.sites.map(site => ({
+        label: site.name,
+        value: site.id,
+        geometry: JSON.parse(site.geometry),
+      }));
+
+      setSelectedProjectStates(sites, project);
+    }
+  }, [selectedProjectId]);
+
+  useEffect(() => {
+    if (selectedProjectSiteId) {
+      const site = projectSites.find(site => site.value === selectedProjectSiteId);
+      if (site) {
+        setSiteCenterCoordinate([]);
+        setSiteBounds(bbox(site.geometry));
+      }
+    }
+  }, [selectedProjectSiteId]);
+
+  const setSelectedProjectStates = (sites: any, project: any) => {
+    setProjectSites(sites);
+    if (sites && sites.length > 0) {
+      setSelectedProjectSiteId(sites[0].value);
+      setSiteCenterCoordinate([]);
+
+      setSiteBounds(bbox(sites[0].geometry));
+    } else {
+      setSelectedProjectSiteId(null);
+      setSiteBounds([]);
+      setSiteCenterCoordinate(turfCenter(JSON.parse(project.geometry)).geometry.coordinates);
```

Comment:
The code also crash here with a JSON.parse error as project.geometry was undefined.